compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUPadDynamicAlloc.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "iree/compiler/Codegen/Utils/Utils.h"
+#include "mlir/Dialect/GPU/Passes.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+static void padAlloc(memref::AllocOp allocOp) {
+  SmallVector<int64_t> shape = llvm::to_vector(allocOp.getType().getShape());
+  SmallVector<OpFoldResult> sizes;
+  size_t dynamicDimIdx = 0;
+  for (int64_t &dimSize : shape) {
+    if (dimSize != ShapedType::kDynamicSize) {
+      Attribute dimAttr = IntegerAttr::get(IndexType::get(allocOp.getContext()),
+                                           APInt(64, dimSize));
+      sizes.push_back(dimAttr);
+      continue;
+    }
+    Value dim = allocOp.dynamicSizes()[dynamicDimIdx++];
+    auto ub = linalg::getConstantUpperBoundForIndex(dim);
+    if (failed(ub)) {
+      return;
+      // allocOp.emitOpError(
+      //    "unexpected allocation without upper bound shapes");
```

Comment:
Should we signal failures and stop the pass if they can't be infered?

Reply:
oops, yes changed it